### PR TITLE
feat: improve onboarding UX & question answer flow

### PR DESCRIPTION
**Pull request by neel-ds**
This PR improves the onboarding flow specifically and optimizes handling user data over bot and mini app. 
- Cleanup email and auth in the first instance
- Show user create on launch (mini app) only if required
- Integrate new user chomp API replacing supabase

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -34,432 +35,409 @@ export const startVercel = async (req: VercelRequest, res: VercelResponse) => {
...
-})
+// https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression
+const emailRegex =
+  /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
 
-bot.action("answering-first-order.2", async ctx =>{
-  ctx.reply(secondPrompt, Markup.inlineKeyboard(formattedSecondButtons))
-})
+bot.hears(emailRegex, async (ctx) => {
+  const { match: emailAddresses } = ctx;
```
Comment (marvinmarnold) [inline src/index.ts]: Shouldn't let the user do this if they have already registered an email.

File: src/index.ts [hunk 2]
```diff
@@ -34,432 +35,409 @@ export const startVercel = async (req: VercelRequest, res: VercelResponse) => {
...
-    console.log(emailAddress)
-    newUser.original_email_address = emailAddress
-    newUser.state = EBotUserState.NEW
-
-    const headers = {
-      Authorization: `Bearer ${process.env.DYNAMIC_TOKEN}`,
-      'Content-Type': 'application/json'
-    }
+bot.action('answering-second-order.10', async (ctx) => {
+  const prompt =
```
Comment (marvinmarnold) [inline src/index.ts]: Would be nice to get rid of this code duplication and refactor logic into helper method.

File: src/index.ts [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
-const supabaseUrl = process.env.SUPABASE_URL!
-const supabaseKey = process.env.SUPABASE_KEY!
-
-const supabase = createClient<Database>(supabaseUrl, supabaseKey)
-// console.log(supabase)
-
 const BOT_TOKEN = process.env.BOT_TOKEN || '';
 const ENVIRONMENT = process.env.NODE_ENV || '';
 
+let user: any;
```
Comment (marvinmarnold) [inline src/index.ts]: I don't think you can use variables like this. In production, the same server instance might not respond to subsequent requests. We should probably use Vercel KV https://vercel.com/docs/storage/vercel-kv or Supabase for storing variables on a per user basis.

**Review by marvinmarnold — CHANGES_REQUESTED** [review]
The comment about variable management is most important.

File: src/index.ts (hunk 1 above)
Comment (neel-ds) [reply src/index.ts]: Validating it before initializing email verification. Fallback mechanism is implemented on both email and OTP hear events so that if user enters similar to that regex, it will act accordingly.

> https://github.com/gator-labs/chomp-bot/blob/111461fb51f32d4e84c1accbd15f1d9f2654865f/src/index.ts#L356-L366

and 

> https://github.com/gator-labs/chomp-bot/blob/111461fb51f32d4e84c1accbd15f1d9f2654865f/src/index.ts#L407-L412

File: src/index.ts (hunk 2 above)
Comment (neel-ds) [reply src/index.ts]: Yes! This part is temporary and it will be replaced with ideal composable code to handle the same requests on answer.

File: src/index.ts (hunk 3 above)
Comment (neel-ds) [reply src/index.ts]: Yes, correct! Redis would be a great option here to manage these variables.

File: package.json [hunk 4]
```diff
@@ -7,13 +7,17 @@
   "homepage": "https://github.com/sollidy/telegram-bot-vercel-boilerplate",
   "dependencies": {
     "@supabase/supabase-js": "^2.43.4",
```
Comment (marvinmarnold) [inline package.json]: You can remove supabase from the project now

File: src/lib/utils.ts [hunk 5]
```diff
@@ -1,18 +1,213 @@
...
+    tg_language_code: ctx.from?.language_code!,
+    state: EBotUserState.NEW,
+    original_email_address: 'overwrite',
+  };
+
+  return newUser;
+};
+
+// Create temporary user with random uuid using Telegram ID
+export const createUserByTelegram = async (tgId: number) => {
```
Comment (marvinmarnold) [inline src/lib/utils.ts]: Can you cast this to a specific custom type like `IChompUser` so its clear what fields are available

**Review by marvinmarnold — COMMENTED** [review]
Can you point me to the code that saves the answers via chomp API?

**Comment by neel-ds** [comment]
@marvinmarnold 
> Can you point me to the code that saves the answers via chomp API?

https://github.com/gator-labs/chomp-bot/blob/5db0fedfba17f323d12540682ac958d82fcc8721/src/index.ts#L175-L213

File: src/lib/utils.ts [hunk 6]
```diff
@@ -1,18 +1,304 @@
...
-        // TODO use chomp API to get an ID
-        // chomp_id: "TODO",
+  const newUser: IBotUser = {
+    tg_id: ctx.from?.id!,
+    tg_first_name: ctx.from?.first_name!,
+    tg_username: ctx.from?.username!,
+    tg_is_bot: ctx.from?.is_bot!,
+    tg_language_code: ctx.from?.language_code!,
+    state: EBotUserState.NEW,
+    original_email_address: 'overwrite',
```
Comment (marvinmarnold) [inline src/lib/utils.ts]: wouldn't it make more sense to make this an optional field with no default value?

File: src/lib/utils.ts [hunk 7]
```diff
@@ -1,18 +1,304 @@
...
+    if (response.data.length <= 0) {
+      return null;
     }
+    return response.data.length;
+  } catch {
+    return null;
+  }
+};
+
+// Initialize email verification process
```
Comment (marvinmarnold) [inline src/lib/utils.ts]: There is unrelated code in this file. Could be broken up into smaller files.

File: .gitignore [hunk 8]
```diff
@@ -11,7 +11,7 @@
 
 # misc
 .DS_Store
-.env*
+.env
```
Comment (marvinmarnold) [inline .gitignore]: A dangerous change. It's normal for devs to create sensitive files like .env.local

I prefer to keep as-is then `git add -f .env.example`

**Review by marvinmarnold — APPROVED** [review]
Some small comments but looks good
